packages/aws-cdk/test/work-graph.test.ts

```diff
@@ -0,0 +1,364 @@
+import { WorkGraph } from '../lib/util/work-graph';
+import { AssetBuildNode, AssetPublishNode, DeploymentState, StackNode } from '../lib/util/work-graph-types';
+
+const DUMMY: any = 'DUMMY';
+
+const sleep = async (duration: number) => new Promise<void>((resolve) => setTimeout(() => resolve(), duration));
+
+// Not great to have actual sleeps in the tests, but they mostly just exist to give the async workflow
+// a chance to start new tasks.
+const SLOW = 200;
+
+/**
+ * Repurposing unused stack attributes to create specific test scenarios
+ * - stack.name          = deployment duration
+ * - stack.displayName   = error message
+ */
+describe('DeployAssets', () => {
```

Comment:
```suggestion
describe('WorkGraph', () => {
```

Reply:
A good reason why I don't like top-level `describe()`s.

They usually end up mirroring the file name, for no additional benefit.